auto merge for Glacier peak

noam987 opened the pull request:
Please Review this to ensure that nothing related to Teleop was accidentally changed.  
The removal of the arm pid is intentional as it is unused anyways

noam987 committed:
First Untested attempt

noam987 committed:
Got Turn to heading Working

noam987 committed:
Got it to move and start up the shooter.
BUG: Anything after the second move command is not scheduled

noam987 committed:
Merge branch 'master' into turnToHeading

noam987 committed:
Finished but did not test basic auton.
Removed the Unused arm Pid.
remvoved the methods that called the arm PID.
Created a Lower Arm Command. (Created as it is needed for auton

noam987 committed:
Added Comment On reset Gyro.
Changed the names of the POV buttons

noam987 committed:
Merge branch 'master' into noam/Auton

noam987 committed:
added more to trench Auton

noam987 committed:
Fixed comments

noam987 committed:
added untuned moveOneMeter and half a meter

ProgrammingSR committed:
merge ToHeading into auton for use

ProgrammingSR committed:
Weird Bug arose with the end of the limelight.
the rest of auton works

noam987 committed:
changes during testing on thursday.
added a stop shooter on disabled and moved over to 3 ball auton.

noam987 committed:
Merge branch 'master' into noam/Auton

noam987 committed:
Mfinishing the merge from master into Noam/Auton.
This code Should be ready for comp

noam987 committed:
Fixed up some comments and unused code

thetaspirit commented on code/src/main/java/frc/robot/Robot.java:
```diff
@@ -172,6 +174,9 @@ public void robotPeriodic() {
   @Override
   public void disabledInit() {
     _robot.disabledInit();
+    new Limelight(_config).setLEDState(1);
```
Move this to robotinit

thetaspirit commented on code/src/main/java/frc/robot/RobotConfig.java:
```diff
@@ -131,10 +131,10 @@
     }
 
     public final class Limelight {
-        public final double kP = .01;
```
Make sure these changes to pid are what you want, even for teleop

thetaspirit commented on code/src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -38,30 +56,71 @@
+        new ShootOneBallCommand(logger, shooter), new ShootOneBallCommand(logger, shooter),
+        new ShootOneBallCommand(logger, shooter), new ShooterShutdownCommand(logger, shooter));
+
+    // TODO finish this auton after Glacier Peak.
```
Remove this for when you merge to master, keep it in the feature branch.   Keep the Todo stub tho

thetaspirit commented on code/src/main/java/frc/robot/commands/navcommands/TrackVisionTarget.java:
```diff
@@ -38,6 +38,7 @@ public TrackVisionTarget(final ILogger logger, RobotConfig config, IDrivetrainSu
     @Override
     public void initialize() {
         super.initialize();
+
```
Remove empty lines added

ProgrammingSR committed:
removed disabled command, set limelight states

ProgrammingSR committed:
code after first two matches of glacier peak

no match time check before deploying climber
auton is working, tuned vision a little bit
drive controller gains edited

ProgrammingSR committed:
code used during our third match

added small integrator to vision tracking

ProgrammingSR committed:
code used during our fourth match

resetting vision tracking pid because we added an integrator last match

ProgrammingSR committed:
code used during our fifth match

removed integrator from vision pid.
added auton that scored in lower port

ProgrammingSR committed:
code used for our sixth match

turned up drive controller scales
reverted auton back to shoot 3 with vision

ProgrammingSR committed:
added D to vision tracking pid, works really well

ProgrammingSR committed:
code used for qual 61

changed heading for climbing
cleaned up magic numbers

ProgrammingSR committed:
add feature to switch direction of winch ratchet

both ratchet servo and climb release servo are set to their locked states are set in robotinit
consider renaming methods and variables

ProgrammingSR committed:
max driver control gains

noam987 commented:
@delaynieym as the driver, what teleop commit do you want into master. Auton is the same for most of them, with the acception being the added move forward auton.
